examples/diffusion_2d.py

```diff
@@ -0,0 +1,195 @@
+state.set_reference_profiles({"b": b_b})
+
+##############################################################################
+# Set up advection schemes
+##############################################################################
+# advection_dict is a dictionary containing field_name: advection class
+ueqn = EulerPoincare(state, u0.function_space())
+supg = True
+if supg:
+    beqn = SUPGAdvection(state, Vb,
+                         supg_params={"dg_direction":"horizontal"},
+                         equation_form="advective")
+else:
+    beqn = EmbeddedDGAdvection(state, Vb,
+                               equation_form="advective")
+
+advection_dict = {}
+#advection_dict["u"] = ThetaMethod(state, u0, ueqn)
+#advection_dict["b"] = SSPRK3(state, b0, beqn)
+
+#For linear equations use:
+#advection_dict["u"] = NoAdvection(state, u0, None) 
+#beqn = LinearAdvection(state, Vb, qbar=b_b)
+#advection_dict["b"] = SSPRK3(state, b0, beqn)
+
+##############################################################################
+# Set up linear solver for the timestepping scheme
+##############################################################################
+linear_solver = IncompressibleSolver(state, L, params=linear_solver_params)
+
+##############################################################################
+# Set up forcing
+##############################################################################
+forcing = IncompressibleForcing(state)
+
+
+##############################################################################
+#Set up diffusion scheme
+##############################################################################
+# mu is a numerical parameter
+# kappa is the diffusion constant for each variable
+# Note that molecular diffusion coefficients were taken from Lautrup, 2005:
+# Kinematic viscosity = 1.*10**(-6)
+# Heat diffusivity = 1.4*10**(-7)
+Vu = u0.function_space()
+Vb = b0.function_space()
+delta = L/columns 		#Grid resolution (same in both directions).
+
+
+bcs_u = [DirichletBC(Vu, 0.0, "bottom"), DirichletBC(Vu, 0.0, "top")]
+#bcs_b = [DirichletBC(Vb, 0.0, "bottom"), DirichletBC(Vb, 0.0, "top")]
+bcs_b = {}
+
+diffusion_dict = {"u": InteriorPenalty(state, Vu, kappa=Constant(1.*10**(-6)),
+                                           mu=Constant(10./delta), bcs=bcs_u),
+                      "b": InteriorPenalty(state, Vb, kappa=Constant(1.4*10**(-7)),
+                                               mu=Constant(10./delta), bcs=bcs_b)}
+
+
+##############################################################################
+# build time stepper
+##############################################################################
+physics_list = {}
+stepper = AdvectionTimestepper(state, advection_dict, diffusion_dict, physics_list)
+
+##############################################################################
+# Run!
+##############################################################################
+stepper.run(t=0, tmax=tmax)
```

Comment:
This whole file looks rather cargo-culted.  Can you cut out all the stuff that isn't actually needed?

It is very confusing that you're doing `AdvectionTimestepper` with no advection, and only diffusion, for example :).